apps/web/app/api/storage/route.ts

```diff
@@ -0,0 +1,334 @@
+import { hasUserEnvironmentAccess } from "@/lib/api/apiHelper";
+import { getServerSession } from "next-auth";
+import { NextRequest, NextResponse } from "next/server";
+import {
+  getFileFromLocalStorage,
+  getFileFromS3,
+  putFileToLocalStorage,
+  putFileToS3,
+} from "@formbricks/lib/services/storage";
+import path from "path";
+
+const UPLOADS_DIR = path.resolve("./uploads");
+
+export async function GET(req: NextRequest) {
+  const searchParams = req.nextUrl.searchParams;
+
+  const fileName = searchParams.get("fileName");
+
+  if (!fileName) {
+    return NextResponse.json(
+      {
+        success: false,
+        message: "Missing required parameters",
+      },
+      {
+        status: 400,
+      }
+    );
+  }
+
+  // parse the fileName to get the environmentId and accessType
+  const fileNameParts = fileName.split("/");
+  const environmentId = fileNameParts[0];
+  const accessType = fileNameParts[1];
+  const baseFileName = fileNameParts[2];
+
+  if (!environmentId || !accessType) {
+    return NextResponse.json(
+      {
+        success: false,
+        message: "Invalid file name",
+      },
+      {
+        status: 400,
+      }
+    );
+  }
+
+  const getFile = async () => {
+    if (
+      !process.env.AWS_ACCESS_KEY ||
+      !process.env.AWS_SECRET_KEY ||
+      !process.env.S3_REGION ||
+      !process.env.S3_BUCKET_NAME
+    ) {
+      try {
+        const { fileBuffer, metaData } = await getFileFromLocalStorage(
+          path.join(UPLOADS_DIR, environmentId, accessType, baseFileName)
+        );
+
+        return new Response(fileBuffer, {
+          headers: {
+            "Content-Type": metaData.contentType,
+            "Content-Disposition": "inline",
+          },
+        });
+      } catch (err) {
+        return NextResponse.json(
+          {
+            success: false,
+            message: "File not found",
+          },
+          {
+            status: 404,
+          }
+        );
+      }
+    }
+
+    try {
+      const { fileBuffer, metaData } = await getFileFromS3(fileName);
+
+      return new Response(fileBuffer, {
+        headers: {
+          "Content-Type": metaData.contentType,
+          "Content-Disposition": "inline",
+        },
+      });
+    } catch (err) {
+      if (err.name === "NoSuchKey") {
+        return NextResponse.json(
+          {
+            success: false,
+            message: "File not found",
+          },
+          {
+            status: 404,
+          }
+        );
+      } else {
+        return NextResponse.json(
+          {
+            success: false,
+            message: "Internal server error",
+          },
+          {
+            status: 500,
+          }
+        );
+      }
+    }
+  };
+
+  if (accessType === "public") {
+    return await getFile();
+  }
+
+  if (accessType !== "private") {
+    return NextResponse.json(
+      {
+        success: false,
+        message: "Invalid access type",
```

Comment:
please provide more information about the allowed access types. see other endpoints for reference.
you can also do the validation with zod if you like / again see other endpoints for an example

Reply:
@mattinannt  done! added zod validation on both access type and file name